File: src/NetCDF.jl
using NCDatasets

jan = Dataset("/Users/arpitasen/Desktop/Sanghavi/Data/EMIT_L1B_RAD_001_20231026T175210_2329912_009.nc")

# Print available variables
println(keys(jan))

# Extract radiance variable
reflectance = jan["radiance"]

# Define band range
start_band = 50 # Oxy - A Band
end_band = 50 # Oxy - A Band

# Select data for specified band range
reflectance_selected_bands = reflectance[start_band:end_band,:,:]
println("Shape of reflectance data for bands $start_band to $(end_band - 1): ", size(reflectance_selected_bands))
println(reflectance_selected_bands)

File: src/NetCDF.py
import netCDF4
from netCDF4 import Dataset

jan = Dataset('/Users/arpitasen/Desktop/Sanghavi/Data/EMIT_L2A_RFL_001_20220903T163129_2224611_012.nc')
reflectance = jan.variables['reflectance']
shape = reflectance.shape
print("Shape of reflectance variable:", shape)
num_bands = shape[2]
print("Number of bands:", num_bands)

print(f"Reflectance data type: {reflectance.dtype}")
print(f"Reflectance dimensions: {reflectance.dimensions}")

band_index = 0
reflectance_band = reflectance[:, :, band_index]
print(f"Reflectance data for band {band_index}:")
print(reflectance_band)

for band in range(num_bands):
    reflectance_band = reflectance[:, :, band]
    print(f"Reflectance data for band {band}:")
    print(reflectance_band)
